protocol/mqtt_paho/v2/protocol.go

```diff
@@ -71,16 +71,35 @@ func (t *Protocol) Send(ctx context.Context, m binding.Message, transformers ...
 	var err error
 	defer m.Finish(err)
 
-	topic := cecontext.TopicFrom(ctx)
-	if topic == "" {
-		topic = t.senderTopic
-	}
-
 	msg := &paho.Publish{
-		QoS:    t.qos,
+		Topic:  t.senderTopic,
 		Retain: t.retained,
-		Topic:  topic,
+		QoS:    t.qos,
+	}
+
+	if cecontext.TopicFrom(ctx) != "" {
+		msg.Topic = cecontext.TopicFrom(ctx)
+	}
+
+	protocolCtx := ProtocolContextFrom(ctx)
+	if protocolCtx != nil {
+		if protocolCtx.Topic != "" {
+			msg.Topic = protocolCtx.Topic
+		}
+		if protocolCtx.Retained != nil {
+			msg.Retain = *protocolCtx.Retained
+		}
+		if protocolCtx.QoS != nil {
+			msg.QoS = *protocolCtx.QoS
+		}
+		if protocolCtx.PacketID != nil {
+			msg.PacketID = *protocolCtx.PacketID
+		}
+		if protocolCtx.PublishProperties != nil {
+			msg.Properties = protocolCtx.PublishProperties
+		}
```

Comment:
Blocking: these are not available in the `New()` constructor call (inconsistent)

Reply:
Yes. A partial constructor is used to handle most cases, and I can add another constructor with full arguments if you think it is necessary.

Reply:
See above